packages/block-library/src/navigation/edit/unsaved-inner-blocks.js

```diff
@@ -51,13 +56,48 @@ export default function UnsavedInnerBlocks( {
 		}
 	}, [ blocks ] );
 
-	// If the current inner blocks object is different in any way
-	// from the original inner blocks from the post content then the
-	// user has made changes to the inner blocks. At this point the inner
-	// blocks can be considered "dirty".
-	// We also make sure the current innerBlocks had a chance to be set.
-	const innerBlocksAreDirty =
-		!! originalBlocks.current && blocks !== originalBlocks.current;
+	let innerBlocksAreDirty = false;
+
+	if (
+		originalBlocks.current &&
+		blocks?.length === 1 &&
+		blocks[ 0 ]?.name === 'core/page-list'
+	) {
+		// If the blocks are a page list, we need to ignore
+		// inner blocks and compare remaining attributes only.
+		// Why? Because Page List block dynamically sets its
+		// child inner blocks async following a REST API request.
+		// This means that the inner blocks are empty when the
+		// block is first inserted, and then populated later once
+		// the REST API request resolves. This causes the original
+		// check to always return dirty as the inner blocks change.
+		// This is a workaround for this specific scenario.
+		const originalPageListBlock = originalBlocks.current[ 0 ];
+		const currentPageListBlock = blocks[ 0 ];
+
+		const {
+			innerBlocks: discardedOriginalInnerBlocks,
+			...originalPageListBlockWithoutInnerBlocks
+		} = originalPageListBlock;
+
+		const {
+			innerBlocks: discardedCurrentInnerBlocks,
+			...currentPageListBlockWithoutInnerBlocks
+		} = currentPageListBlock;
+
+		innerBlocksAreDirty = ! isEqual(
+			originalPageListBlockWithoutInnerBlocks,
+			currentPageListBlockWithoutInnerBlocks
+		);
+	} else {
+		// If the current inner blocks object is different in any way
+		// from the original inner blocks from the post content then the
+		// user has made changes to the inner blocks. At this point the inner
+		// blocks can be considered "dirty".
+		// We also make sure the current innerBlocks had a chance to be set.
```

Comment:
```suggestion
		// If the current inner blocks object does not display referential equality
		// with the original inner blocks from the post content then the
		// user has made changes to the inner blocks. At this point the inner
		// blocks can be considered "dirty".
		// We also make sure the current innerBlocks had a chance to be set.
```

Let's be specific here.